salt/cloud/__init__.py

```diff
@@ -1183,11 +1171,6 @@ def create(self, vm_, local_master=True):
             'minion', vm_, self.opts, default={}
         )
 
-        # Since using "provider: <provider-engine>" is deprecated, alias provider
-        # to use driver: "driver: <provider-engine>"
-        if 'provider' in vm_:
-            vm_['driver'] = vm_.pop('provider')
-
```

Comment:
There is an issue here: `provider.pop('provider')` returns a result that can be split in two parts:

    > /home/eradman/local/saltenv/lib/python2.7/site-packages/salt/cloud/__init__.py(1192)create()
    -> alias, driver = vm_['driver'].split(':')
    (Pdb) vm_['driver']
    'rad_joyent_account:joyent'

Whereas after this change `vm_['driver']` only contained the string with the driver

    > /home/eradman/local/saltenv/lib/python2.7/site-packages/salt/cloud/__init__.py(1175)create()
    -> alias, driver = vm_['driver'].split(':')
    (Pdb) vm_['driver']
    'joyent'